cpp/arcticdb/python/python_to_tensor_frame.cpp

```diff
@@ -68,19 +68,34 @@ NativeTensor obj_to_tensor(PyObject *ptr) {
             auto none = py::none{};
             auto obj = reinterpret_cast<PyObject **>(arr->data);
             bool empty = false;
+            bool all_nans = false;
             PyObject *sample = *obj;
             if (sample == none.ptr() || is_py_nan(sample)) {
-                // Iterate till we find the first non null element, the frontend ensures there is at least one.
+                empty = true;
+                all_nans = true;
                 util::check(c_style, "Non contiguous columns with first element as None not supported yet.");
-                auto casted_col =
-                    std::find_if(obj, obj + size, [&none](auto val) { return val != none.ptr() && !is_py_nan(val); });
-                empty = casted_col == obj + size;
-                sample = *casted_col;
+                PyObject** current_object = obj;
+                while(current_object < obj + size) {
+                    if(*current_object == none.ptr()) {
+                        all_nans = false;
+                    } else if(is_py_nan(*current_object)) {
+                        empty = false;
+                    } else {
+                        all_nans = false;
+                        empty = false;
+                        break;
+                    }
+                    ++current_object;
+                }
+                sample = *current_object;
             }
-            if (empty || is_unicode(sample)) {
+            if (empty) {
+                val_type = ValueType::EMPTY;
+            } else if(all_nans || is_unicode(sample)){
                 val_type = ValueType::UTF_DYNAMIC;
```

Comment:
Could you document why `UTF_DYNAMIC` is used if all values are nans (`all_nans`)?

Reply:
To be fair I'm not 100% sure why UTF_DYNAMIC was chosen. Could be related to the size of `NaN` on disc. @alexowens90 @willdealtry can you confirm?

Reply:
This is unfortunately a historic thing where NaNs were used to indicate missing strings. I'm not going to defend it making sense but there's not a lot we can do about it now. The decision was made that this unfortunate fact should go no further, so any other missing Python type, like a decimal or an array, should be represented by None.

Might be worth a comment.

Reply:
I think the question is not why we treat `NaN` as as string (I think we've already got that covered), but why we set the DataType to be `UTF_DYNAMIC` and not `ASCII_DYNAMIC` for example.